modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/AdapterConfig.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018-2023 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.snowplow.enrich.common.adapters
+
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaVer}
+
+/**
+ * Configuration for an adapter.
+ * @param schemas optional map of vendor name to event name to schema URI
+ */
+case class AdapterConfig(schemas: Option[Map[String, Map[String, String]]] = None) {
```

Comment:
Before your changes, we were always using hard coded schemas in the adapters. Now we make it possible to override them. The way we usually do this is to remove entirely the hard coded values from the code, and put all the defaults in `src/main/resources/application.conf`.

This could look something like
```
"adapters": {
  "callrail": {
    "call_complete": "iglu:com.callrail/call_complete/jsonschema/1-0-2"
  },
  "googleAnalytics": {
    ...,
    "screenview": "iglu:com.google.analytics/screenview/jsonschema/1-0-0",
    "transaction": "iglu:com.google.analytics/transaction/jsonschema/1-0-0"
  },
  ...
}
````
The main difficulty with this approach is that at the moment the defaults are only per app (e.g. for [enrich-kinesis](https://github.com/snowplow/enrich/blob/master/modules/kinesis/src/main/resources/application.conf)), there is no default in [common/src/resources/](https://github.com/snowplow/enrich/tree/master/modules/common/src), so we'd need to add one there and make sure that it is resolved by the apps when loading the configuration.

If this is too complicated, I think that we should at least organize the schemas by adapter in the config, e.g.
```
"adapters": {
  "callrail": {
    "schemas": {
      { "com.callrail" : { "call_complete": "iglu:com.callrail/call_complete/jsonschema/1-0-2" }}
    }
  },
  "googleAnalytics" : {
    "schemas": {
      { "com.google.analytics": {
        "event": "iglu:com.google.analytics/event/jsonschema/1-0-0" },
        "screen_view": "iglu:com.google.analytics/screen_view/jsonschema/1-0-0"}
      }
    }
  }
}
```
and then pass the appropriate schemas when instantating an adapter, e.g.
```
new CallrailAdapter(adapterConfig.callrail)
```

Reply:
Hi @benjben, thanks for the feedback!

I have changed it to only store the defaults in the configuration and added a `application-default.conf` file to `common/src/resources` with the default values.

The problem is that this resulted in quite a bit of duplication of the configuration:

1. the values are now present in all the `extended` config files.
2. I had to make them also available in the tests, which I did by adding them to the `SpecHelpers`. However since there is a different `SpecHelpers` for each test module, I had to duplicate this too.

If you have any suggestions or ideas how to prevent the duplication, I would be happy to change it. I especially don't like the duplication in the `SpecHelpers` but couldn't find a different way to do it.

Reply:
I have made the changes as we discussed (remove the config from extended conf files and share the same `SpecHelpers` defaults) and it helped a lot with the config duplication!